### PR TITLE
fix(replay-onboarding): Adjust Replay onboarding instructions for Astro

### PR DESCRIPTION
**Pull request by Lms24**
This PR adjusts the in-product Replay onboarding doc by:

- stating that Replay is enabled by default in the Astro SDK (=> configuration becomes optional)
- diffrentiating between which options can be set in the simple `astro.config.mjs` vs which options can only be set by creating a dedicated `sentry.client.config.js` file. 
- Adjusting text as necessary.

Before (taken from https://github.com/getsentry/sentry-javascript/issues/12408): 

image

After:
image

closes https://github.com/getsentry/sentry-javascript/issues/12408

cc @whitep4nth3r

### CODE REVIEW

**Review by mydea — APPROVED** [review]
hmm, should we also document/show how to deactivate replay? If users do not have replay selected in the onboarding, we should show some config that removes it, I guess? (Or is this already happening?)

**Comment by Lms24** [comment]
> Also — why would we include the replay configuration in both files? Surely it makes more sense just to include it in the client config?

We don't. The idea is:
- either you use the simpe `astro.config.mjs` config which has [limited options](https://docs.sentry.io/platforms/javascript/guides/astro/manual-setup/#astro-integration-options). This is the default experience. You can control replay with the replay sample rates. If both are 0, replay is not enabaled, otherwise the integration is added. By default we sample with 0.1 on session and 1.0 on error.
- or you use `sentry.client.config.js` which works like any other `Sentry.init` call. This however, is opt-in. Once you created this file, the configuration in `astro.config.mjs` has no effect anymore (with the exception of source maps upload).

So this updated guide tries to accommodate both of these use cases

> This makes more sense, but still doesn't explain why I see no replays after a week now 😅

Hmm nope 😅 Do you see debug logs indicating if Replay was installed or not? do you have a link to your project repo by chance?

**Comment by whitep4nth3r** [comment]
Conversation was continued in Slack and I got it set up. Essentially we've all concluded that we need some clearer instructions around configuring Astro, given that you can use three files to configure things and some settings might override the others.

File: static/app/gettingStartedDocs/javascript/astro.tsx [hunk 1]
```diff
@@ -206,20 +203,74 @@ const onboarding: OnboardingConfig = {
 };
 
 const replayOnboarding: OnboardingConfig = {
-  install: () => getInstallConfig(),
+  install: () => [
+    {
+      ...getInstallConfig()[0],
+      additionalInfo:
+        'Session Replay is enabled by default when you install the Astro SDK!',
```
Comment (michellewzhang) [inline static/app/gettingStartedDocs/javascript/astro.tsx:210]: do we need to state that it's enabled by default twice? it's here and also on the first line of the configure step

File: static/app/gettingStartedDocs/javascript/astro.tsx [hunk 2]
```diff
@@ -206,20 +203,74 @@ const onboarding: OnboardingConfig = {
...
+          code: <code />,
+          code2: <code />,
+          link: (
+            <ExternalLink
+              href={
+                'https://docs.sentry.io/platforms/javascript/guides/astro/session-replay/'
+              }
+            />
+          ),
+        }
```
Comment (michellewzhang) [inline static/app/gettingStartedDocs/javascript/astro.tsx:228]:
```suggestion
        {
          link: (
            <ExternalLink
              href={
                'https://docs.sentry.io/platforms/javascript/guides/astro/session-replay/'
              }
            />
          ),
        }
```

File: static/app/gettingStartedDocs/javascript/astro.tsx (hunk 1 above)
Comment (Lms24) [reply static/app/gettingStartedDocs/javascript/astro.tsx:210]: Yeah I was thinking better twice than confuse people why there's the configure section but you're right, let's leave it here and remove it from the configure section.

File: static/app/gettingStartedDocs/javascript/astro.tsx [hunk 3]
```diff
@@ -206,20 +203,74 @@ const onboarding: OnboardingConfig = {
...
+            }
+          ),
+        },
         {
           code: [
             {
               label: 'JavaScript',
               value: 'javascript',
               language: 'javascript',
+              filename: 'sentry.client.config.js',
```
[the remark this answers is not in the document]
Comment (Lms24) [reply static/app/gettingStartedDocs/javascript/astro.tsx:273]: Yup, good suggestion! I added this:
image

**Comment by Lms24** [comment]
Thx for the reviews! I applied the changes. I'm on vacation though for the next two weeks so please feel free to merge this :)

**Review by michellewzhang — APPROVED** [review]
lgtm! thanks for making the changes!

**Review by ryan953 — APPROVED** [review]

**Comment by ryan953** [comment]
> Thx for the reviews! I applied the changes. I'm on vacation though for the next two weeks so please feel free to merge this once you think this is good to go :)

@michellewzhang do you want to click merge if you've got the bandwidth to support it?

**Comment by michellewzhang** [comment]
> > Thx for the reviews! I applied the changes. I'm on vacation though for the next two weeks so please feel free to merge this once you think this is good to go :)
> 
> 
> 
> @michellewzhang do you want to click merge if you've got the bandwidth to support it? 

yup, i'll merge and watch over

**Review by michellewzhang — APPROVED** [review]

**Pull request merged**